### DISCUSSION TITLE
Unofficial Additions

### DISCUSSION THREAD

**Discussion by vhscom** [rgossiaux/svelte-headlessui#82] [Ideas]
Given this library is officially unofficial it could serve as a good place to make some officially unofficial additions to the headless component gallery to make it even more awesome. What are some of the most sought after headless components which could add value?

Here's one to start: A date picker a la https://github.com/duetds/date-picker

**Comment by rgossiaux** [comment]
I intend on maintaining this project as a straight port of Tailwind's Headless UI library, so I think other components are best served in another repository. The worst case would be if I added something custom myself, Tailwind later added the same component in a different way, and the two were not compatible. Adding new components is also a lot of work to test, document, and maintain them at the same quality bar as the rest of the library. There's not too much downside to a separate repository, by comparison.

Personally there are two different things I've been thinking about in this space:

* Using this library to form the backbone of components for some existing CSS frameworks or design systems. I've prototyped putting Daisy UI on top of these components, for example, and intend on polishing & sharing that in some form in the not-too-distant future.
* Porting some subset of react-aria as Svelte actions. I think this has the potential to be pretty interesting.

But I still have some more improvements to this library to ship first (besides the things going on in the rest of my life 😄 ) so those are taking a back seat for now.

**Reply by vhscom** [reply]
> The worst case would be if I added something custom myself, Tailwind later added the same component in a different way, and the two were not compatible.

Yeah I was thinking of that after I wrote this. Completely agree addons would be best left to another project. Now I suppose we just need a way to figure out how to do that. A pattern for creating headless components which could be reused in various projects.
